src/control/system/raft/database.go

```diff
@@ -898,14 +907,41 @@ func (db *Database) FindPoolServiceByLabel(label string) (*system.PoolService, e
 	return nil, system.ErrPoolLabelNotFound(label)
 }
 
+func (db *Database) TakePoolLock(ctx context.Context, uuid uuid.UUID) (*PoolLock, error) {
```

Comment:
Needs documentation comment

Reply:
Thanks, missed this. Will add it if there is any other feedback that requires another push, or in a follow-up. I'd like to get this landed to resolve some intermittent failures that we're still seeing.